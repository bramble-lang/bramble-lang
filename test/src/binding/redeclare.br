fn my_main() {
    x: i32 := 5;
    x: i32 := 10;
    return;
}