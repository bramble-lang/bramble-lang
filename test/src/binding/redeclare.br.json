[{"id": 1, "stage": "lexer", "source": [0, 2], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [3, 10], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [10, 11], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [13, 15], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [16, 19], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [20, 21], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [26, 29], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [30, 31], "ok": "Identifier"},
{"id": 10, "stage": "lexer", "source": [31, 32], "ok": "Operator"},
{"id": 11, "stage": "lexer", "source": [33, 36], "ok": "Primitive"},
{"id": 12, "stage": "lexer", "source": [37, 39], "ok": "Operator"},
{"id": 13, "stage": "lexer", "source": [40, 41], "ok": "Integer"},
{"id": 14, "stage": "lexer", "source": [41, 42], "ok": "Operator"},
{"id": 15, "stage": "lexer", "source": [47, 50], "ok": "Keyword"},
{"id": 16, "stage": "lexer", "source": [51, 52], "ok": "Identifier"},
{"id": 17, "stage": "lexer", "source": [52, 53], "ok": "Operator"},
{"id": 18, "stage": "lexer", "source": [54, 57], "ok": "Primitive"},
{"id": 19, "stage": "lexer", "source": [58, 60], "ok": "Operator"},
{"id": 20, "stage": "lexer", "source": [61, 63], "ok": "Integer"},
{"id": 21, "stage": "lexer", "source": [63, 64], "ok": "Operator"},
{"id": 22, "stage": "lexer", "source": [69, 75], "ok": "Keyword"},
{"id": 23, "stage": "lexer", "source": [76, 77], "ok": "Integer"},
{"id": 24, "stage": "lexer", "source": [77, 78], "ok": "Operator"},
{"id": 25, "stage": "lexer", "source": [79, 80], "ok": "Operator"},
{"id": 31, "parent_id": 29, "stage": "parser", "source": [16, 19], "ok": "Primitive Type"},
{"id": 29, "parent_id": 28, "stage": "parser", "source": [0, 19], "ok": "Routine Declaration"},
{"id": 35, "parent_id": 34, "stage": "parser", "source": [33, 36], "ok": "Primitive Type"},
{"id": 34, "parent_id": 33, "stage": "parser", "source": [30, 36], "ok": "Identifier Declaration"},
{"id": 46, "parent_id": 42, "stage": "parser", "source": [40, 41], "ok": "Number"},
{"id": 42, "parent_id": 41, "stage": "parser", "source": [40, 41]},
{"id": 41, "parent_id": 40, "stage": "parser", "source": [40, 41]},
{"id": 40, "parent_id": 39, "stage": "parser", "source": [40, 41]},
{"id": 39, "parent_id": 38, "stage": "parser", "source": [40, 41]},
{"id": 38, "parent_id": 37, "stage": "parser", "source": [40, 41]},
{"id": 37, "parent_id": 33, "stage": "parser", "source": [40, 41]},
{"id": 33, "parent_id": 32, "stage": "parser", "source": [26, 41], "ok": "Let Binding"},
{"id": 32, "parent_id": 28, "stage": "parser", "source": [26, 42], "ok": "Statement Bind"},
{"id": 50, "parent_id": 49, "stage": "parser", "source": [54, 57], "ok": "Primitive Type"},
{"id": 49, "parent_id": 48, "stage": "parser", "source": [51, 57], "ok": "Identifier Declaration"},
{"id": 61, "parent_id": 57, "stage": "parser", "source": [61, 63], "ok": "Number"},
{"id": 57, "parent_id": 56, "stage": "parser", "source": [61, 63]},
{"id": 56, "parent_id": 55, "stage": "parser", "source": [61, 63]},
{"id": 55, "parent_id": 54, "stage": "parser", "source": [61, 63]},
{"id": 54, "parent_id": 53, "stage": "parser", "source": [61, 63]},
{"id": 53, "parent_id": 52, "stage": "parser", "source": [61, 63]},
{"id": 52, "parent_id": 48, "stage": "parser", "source": [61, 63]},
{"id": 48, "parent_id": 47, "stage": "parser", "source": [47, 63], "ok": "Let Binding"},
{"id": 47, "parent_id": 28, "stage": "parser", "source": [47, 64], "ok": "Statement Bind"},
{"id": 87, "parent_id": 83, "stage": "parser", "source": [76, 77], "ok": "Number"},
{"id": 83, "parent_id": 82, "stage": "parser", "source": [76, 77]},
{"id": 82, "parent_id": 81, "stage": "parser", "source": [76, 77]},
{"id": 81, "parent_id": 80, "stage": "parser", "source": [76, 77]},
{"id": 80, "parent_id": 79, "stage": "parser", "source": [76, 77]},
{"id": 79, "parent_id": 78, "stage": "parser", "source": [76, 77]},
{"id": 78, "parent_id": 77, "stage": "parser", "source": [76, 77]},
{"id": 77, "parent_id": 28, "stage": "parser", "source": [69, 77], "ok": "Return"},
{"id": 28, "parent_id": 26, "stage": "parser", "source": [0, 80], "ok": "Funtion Definition"},
{"id": 26, "stage": "parser", "source": [0, 80], "ok": "File Module"},
{"id": 90, "stage": "canonize-item-path", "source": [0, 80], "ok": "$redeclare::redeclare"},
{"id": 91, "stage": "canonize-item-path", "source": [0, 80], "ok": "$redeclare::redeclare::redeclare"},
{"id": 92, "stage": "canonize-item-path", "source": [0, 80], "ok": "$redeclare::redeclare::my_main"},
{"id": 93, "stage": "type-resolver", "source": [40, 41], "ok": "i64"},
{"id": 94, "stage": "type-resolver", "source": [26, 42], "ok": "i64"},
{"id": 95, "stage": "type-resolver", "source": [61, 63], "ok": "i64"},
{"id": 96, "stage": "type-resolver", "source": [47, 64], "error": "L3: x already declared"}]