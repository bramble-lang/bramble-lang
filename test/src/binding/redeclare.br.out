Error: L3: x already declared
